ext/mysqlnd/mysqlnd_result.c

```diff
@@ -252,7 +251,11 @@ mysqlnd_query_read_result_set_header(MYSQLND_CONN_DATA * conn, MYSQLND_STMT * s)
 				enum_mysqlnd_collected_stats statistic = STAT_LAST;
 
 				DBG_INF("Result set pending");
-				SET_EMPTY_MESSAGE(conn->last_message.s, conn->last_message.l);
+				if (conn->last_message.s) {
+					mnd_efree(conn->last_message.s);
+					conn->last_message.s = NULL;
+				}
+				conn->last_message.l = 0;
```

Comment:
An alternative would be `mysqlnd_set_message(conn->last_message, NULL, 0)` here. Might be nicer to avoid touching internals?